src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -118,38 +131,73 @@ public void addPullRequestReviewer(String project, String repository, String pul
   }
   
   /**
-   * Post one comment by found issue on Stash.
+   * Push SonarQube report into the pull-request as comments.
    */
-  public void postCommentPerIssue(String project, String repository, String pullRequestId, String sonarQubeURL, SonarQubeIssuesReport issueReport, StashDiffReport diffReport, StashClient stashClient){
+  public void postSonarQubeReport(String project, String repository, String pullRequestId, String sonarQubeURL, SonarQubeIssuesReport issueReport, StashDiffReport diffReport, StashClient stashClient) {
     try {
-      // to optimize request to Stash, builds comment match ordered by filepath
-      Map<String,StashCommentReport> commentsByFile = new HashMap<>();
-      for (SonarQubeIssue issue : issueReport.getIssues()) {
-        if (commentsByFile.get(issue.getPath()) == null){
-          StashCommentReport comments = stashClient.getPullRequestComments(project, repository, pullRequestId, issue.getPath());
+      postCommentPerIssue(project, repository, pullRequestId, sonarQubeURL, issueReport.getIssues(), diffReport, stashClient);
+      
+      LOGGER.info("New SonarQube issues have been reported to Stash.");
+      
+    } catch (StashClientException e){
+      LOGGER.error("Unable to link SonarQube issues to Stash: {}", e.getMessage());
+      LOGGER.debug("Exception stack trace", e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Exception stack trace" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)